src/main/java/de/cas/mse/exercise/SumDigPower.java

```diff
@@ -5,37 +5,44 @@
 
 public class SumDigPower {
 
-	public List<Long> sumDigPow(long a, long b) {
+	public List<Long> getEurekaNumbers(long intervalStart, long intervalEnd) {
 
-		List<Long> result = new ArrayList<Long>();
+		List<Long> eurekaNumbers = new ArrayList<Long>();
 
-		for (long i = a; i < b; i++) {
+		for (long currentNumber = intervalStart; currentNumber < intervalEnd; currentNumber++) {
 
-			List<Long> longs = new ArrayList<Long>();
+			List<Long> digits = splitInDigits(currentNumber);
 
-			String temp = Long.toString(i);
+			long sum = sumExponents(digits);
 
-			// Split values
-			for (int j = 0; j < temp.length(); j++) {
-				longs.add(Long.valueOf(temp.substring(j, j + 1)));
-			}
-
-			// Create sum
-			long sum = 0;
-			for (int j = 1; j <= longs.size(); j++) {
-				sum += Math.pow(longs.get(j - 1), j);
-			}
-
-			// Test if sum is equal
-			if (sum == i) {
-				result.add(sum);
+			if (sum == currentNumber) {
+				eurekaNumbers.add(sum);
 			}
 
 		}
 
-        System.out.println(result);
-		return result;
-
+        System.out.println(eurekaNumbers);
+		return eurekaNumbers;
 	}
 
+    private List<Long> splitInDigits(long currentNumber) {
+        List<Long> digits = new ArrayList<Long>();
+
+        String currentNumberAsString = Long.toString(currentNumber);
+
+        for (int j = 0; j < currentNumberAsString.length(); j++) {
+        	digits.add(Long.valueOf(currentNumberAsString.substring(j, j + 1)));
+        }
+        return digits;
+    }
+
+    private long sumExponents(List<Long> digits) {
+        long sum = 0;
+        for (int power = 1; power <= digits.size(); power++) {
```

Comment:
maybe: index as iteration variable and power as index +1

Reply:
and for the method name: maybe power would be better?